### PR TITLE
Fix/rgroup sdf isotope

### PR DESCRIPTION
**Pull request by bp-kelley**
Fixes:

 (1) Add better autodetection of labels
 (2) add dummyatom label (for *CC style queries)
 (3) don't fall back to indices when onlyMatchAtRgroups is set

The autodetection rules are now all or nothing when searching for labels.  The following order is used:

   1. MDLRGroup
   2. Atom Map Number
   3. isotope
   4. dummy atom (*) [n.b. if dummy atoms are used, relabeling is automatic]
   5. indices (only if onlyMatchAtRGroups == false)
   6. if we get here, the core is rejected.

### CODE REVIEW

File: Docs/Book_jp/conf.py [hunk 1]
```diff
@@ -137,9 +137,9 @@
 # built documents.
 #
 # The short X.Y version.
-version = '1.00'
+version = '2019.03'
```
Comment (bp-kelley) [inline Docs/Book_jp/conf.py]: I have on idea how this got in here.

**Comment by bp-kelley** [comment]
@greglandrum The atom coord regression test is failing now (coord.z != 0) I can't see where anything I did might affect this.  Could you take a look?

File: Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp [hunk 2]
```diff
@@ -56,10 +56,48 @@ const std::string RLABEL = "tempRlabel";
 const std::string SIDECHAIN_RLABELS = "sideChainRlabels";
 const std::string done = "RLABEL_PROCESSED";
 
+// Various places where rgroups can be labeled
+//  the order of precedence
+const std::string RGROUP_LABELS = "RGroupLabels";
```
Comment (greglandrum) [inline Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp]: Wouldn't these be better as an enum?

File: Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp [hunk 3]
```diff
@@ -496,7 +597,7 @@ struct RGroupDecompData {
     size_t idx = 0;
     for (auto coreIt = cores.begin(); coreIt != cores.end(); ++coreIt, ++idx) {
       RWMol *alignCore = coreIt->first ? &cores[0] : nullptr;
-      params.prepareCore(coreIt->second, alignCore);
+	  PRECONDITION(params.prepareCore(coreIt->second, alignCore), "Could not prepare at least one core");
```
Comment (greglandrum) [inline Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp]: This isn't really a PRECONDITION. 
To be consistent it should be `CHECK_INVARIANT`

File: Code/GraphMol/RGroupDecomposition/testRGroupDecomp.cpp [hunk 4]
```diff
@@ -55,7 +55,7 @@ void CHECK_RGROUP(RGroupRows::const_iterator &it, std::string expected,
   }
   std::string result = str.str();
 
-  if (expected != result) {
+  if (1 || expected != result) {
```
Comment (greglandrum) [inline Code/GraphMol/RGroupDecomposition/testRGroupDecomp.cpp]: Probably don't intend for this `1 ||` to still be here

File: Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp (hunk 2 above)
Comment (bp-kelley) [reply Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp]: I had them as strings for debugging, but an enum would work just fine.

Comment (greglandrum) [reply Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp]: Worth making the change pre-merge or would you rather just leave it?

**Comment by bp-kelley** [comment]
I’ll try and have this sorted this week along with the new oddity you discovered.

**Pull request merged**